tests/hooks/utils/conversion-utils.spec.ts

```diff
@@ -1,35 +1,93 @@
 import fs from "fs";
 import path from "path";
-import { describe, expect, test } from "vitest";
+import { beforeEach, describe, expect, test, vi } from "vitest";
 import { writeJSONToFile } from "../../../hooks/utils/conversion-utils";
 
 describe("writeJSONToFile", () => {
-  test("write a dummy json to a file", async () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+    vi.mock("fs");
+  });
+
+  test("writes a file in an nonexistent directory", async () => {
+    fs.promises.stat = vi.fn().mockImplementation(() => {
+      throw new Error("Mocked error");
+    });
+    fs.promises.mkdir = vi.fn().mockImplementation(() => true);
+    fs.promises.writeFile = vi.fn().mockImplementation(() => true);
+
     const mock = [
       {
-        name: "Main",
-        description: "The main Qiskit packages maintained by IBM Quantum",
-      },
-      {
-        name: "Extensions",
-        description: "IBM Quantum supported Qiskit extensions",
+        name: "Scottie Wuckert",
+        image: "/images/advocates/downloaded/rec8zXNiryVq3Q6kh.jpg",
+        region: "America (North)",
+        city: "New York",
+        country: ["United States"],
+        slackId: "U017XULSMUY",
+        slackUsername: "Maria.Dickens34",
       },
       {
-        name: "Community",
-        description:
-          "Software packages supported by the Qiskit community, not maintained by IBM Quantum",
+        name: "Abeer Vaishnav",
```

Comment:
It's a good idea to avoid real names in mock data.

Some _safe_ names to use are **John Doe** and **Jane Doe** for instance: https://en.wikipedia.org/wiki/John_Doe

Reply:
I think now all names are replaced by Mr and Mrs Doe